test/test_torch.py

```diff
@@ -6059,11 +6064,11 @@ def test_inverse_helper(matrix, batches, n):
             # large batch size and large matrix size will be tested in test_inverse_many_batches (slow test)
             if batches and batches[0] == 32 and n == 256:
                 continue
-            _matrices = random_fullrank_matrix_distinct_singular_value(n, *batches).to(device)
+            _matrices = random_fullrank_matrix_distinct_singular_value(n, *batches, dtype=dtype, device=device)
```

Comment:
`random_fullrank_matrix_distinct_singular_value` uses `svd`, and currently `svd` is extremely slow on GPU. Let's keep this calculated on CPU then `to(device)` for now. Same below.